### PR TITLE
refresh parent to enable compatability testing with recent jenkins versions

### PR DESCRIPTION
**Pull request by jtnord**
Cleanup dependency declarations

pretty much obsoletes every other PR with the exception of #149 (which is a bump of the only directly bundled library)

- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,18 +14,14 @@
   <packaging>hpi</packaging>
 
   <properties>
-<!--    <jenkins.version>2.190.3</jenkins.version>-->
-    <jenkins.version>2.277.4</jenkins.version>
-    <java.level>8</java.level>
```
Comment (jtnord) [inline pom.xml:19]: java level is now defined by the version of jenkins being targeted [ref](https://github.com/jenkinsci/plugin-pom/releases/tag/plugin-4.40)

File: pom.xml [hunk 2]
```diff
@@ -14,18 +14,14 @@
...
+    <jenkins.version>2.289.3</jenkins.version>
     <slf4jVersion>1.7.26</slf4jVersion>
   </properties>
 
   <name>CloudBees CD</name>
   <description>CloudBees CD Application Release Orchestration</description>
   <url>https://github.com/jenkinsci/electricflow-plugin</url>
 
-  <!-- The default licence for Jenkins OSS Plugins is MIT. Substitute for the applicable one if needed. -->
```
Comment (jtnord) [inline pom.xml:27]: remove comment that was leftover from project creatin

File: pom.xml [hunk 3]
```diff
@@ -65,48 +61,26 @@
     <dependencies>
       <dependency>
         <groupId>io.jenkins.tools.bom</groupId>
-        <artifactId>bom-2.190.x</artifactId>
-        <version>9</version>
+        <artifactId>bom-2.289.x</artifactId>
```
Comment (jtnord) [inline pom.xml:64]: align bom with Jenkins version

File: pom.xml [hunk 4]
```diff
@@ -65,48 +61,26 @@
...
         <type>pom</type>
       </dependency>
     </dependencies>
   </dependencyManagement>
   <dependencies>
-<!--    <dependency>-->
-<!--      <groupId>commons-io</groupId>-->
-<!--      <artifactId>commons-io</artifactId>-->
-<!--      <version>2.8.0</version>-->
-<!--    </dependency>-->
```
Comment (jtnord) [inline pom.xml:80]: was commented out - just removed it

File: pom.xml [hunk 5]
```diff
@@ -65,48 +61,26 @@
...
-    <dependency>
-      <groupId>org.apache.maven.shared</groupId>
-      <artifactId>maven-shared-utils</artifactId>
-      <version>3.1.0</version>
-    </dependency>
-    <dependency>
-      <groupId>org.codehaus.plexus</groupId>
-      <artifactId>plexus-utils</artifactId>
-      <version>3.3.0</version>
-    </dependency>
```
Comment (jtnord) [inline pom.xml:90]: dependencies where unused

File: pom.xml [hunk 6]
```diff
@@ -65,48 +61,26 @@
...
     <dependency>
-      <groupId>org.apache.sshd</groupId>
-      <artifactId>sshd-core</artifactId>
-      <version>1.7.0</version>
```
Comment (jtnord) [inline pom.xml:94]: sshd is now a module - but at any rate it was unused.
obsoletes #159 & #294

File: pom.xml [hunk 7]
```diff
@@ -65,48 +61,26 @@
...
+      <groupId>org.owasp.encoder</groupId>
+      <artifactId>encoder</artifactId>
+      <version>1.2.2</version>
```
Comment (jtnord) [inline pom.xml:75]: this is not new just moved the only compile scope non jenkins plugin dependency to the top.

File: pom.xml [hunk 8]
```diff
@@ -115,54 +89,41 @@
...
     <dependency>
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>variant</artifactId>
-      <version>1.2</version>
     </dependency>
     <dependency>
       <groupId>org.jenkins-ci.plugins.pipeline-stage-view</groupId>
-      <artifactId>pipeline-stage-view</artifactId>
-      <version>2.12</version>
-      <type>hpi</type>
```
Comment (jtnord) [inline pom.xml:148]: unused by the plugin.

File: pom.xml [hunk 9]
```diff
@@ -115,54 +89,41 @@
...
+      <artifactId>pipeline-rest-api</artifactId>
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.jenkins-ci.plugins.pipeline-stage-view</groupId>
-      <artifactId>pipeline-rest-api</artifactId>
-      <version>2.12</version>
-      <type>hpi</type>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-job</artifactId>
```
Comment (jtnord) [inline pom.xml:120]: was used but undeclared relying on this transitively :-o

File: pom.xml [hunk 10]
```diff
@@ -177,19 +138,17 @@
...
       <artifactId>mockito-inline</artifactId>
-      <version>3.5.13</version>
       <scope>test</scope>
     </dependency>
     <dependency>
       <groupId>com.github.tomakehurst</groupId>
-      <artifactId>wiremock-standalone</artifactId>
-      <version>2.27.2</version>
+      <artifactId>wiremock-jre8-standalone</artifactId>
+      <version>2.33.2</version>
```
Comment (jtnord) [inline pom.xml:151]: bumped the version to the latest jdk8 version whilst we are here.

File: src/main/java/org/jenkinsci/plugins/electricflow/CloudBeesFlowCallRestApiStep.java [hunk 11]
```diff
@@ -9,7 +9,7 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Set;
-import javax.annotation.Nonnull;
+import edu.umd.cs.findbugs.annotations.NonNull;
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/electricflow/CloudBeesFlowCallRestApiStep.java:12]: jsr-305 is dead long live spotbugs :)

File: src/main/java/org/jenkinsci/plugins/electricflow/extension/impl/CloudBeesFlowPipelineRest.java [hunk 12]
```diff
@@ -9,7 +9,7 @@
 import org.jenkinsci.plugins.variant.OptionalExtension;
 import org.jenkinsci.plugins.workflow.job.WorkflowRun;
 
-@OptionalExtension(requirePlugins = "pipeline-rest-api")
+@OptionalExtension(requirePlugins = {"pipeline-rest-api","workflow-job"})
```
Comment (jtnord) [inline src/main/java/org/jenkinsci/plugins/electricflow/extension/impl/CloudBeesFlowPipelineRest.java:12]: https://github.com/jenkinsci/electricflow-plugin/pull/311/files#diff-5fe4dcab045ed4e397c90b2207b14cf8d791b70c7a5f5db9cab0255acf95244eR19 used `WorkflowRun` yet the dependency was not added here relying on the transitive dependency of `pipeline-rest-api`.

**Review by basil — APPROVED** [review]

**Review by ovasilko — APPROVED** [review]

**Review by justnoxx — APPROVED** [review]

**Pull request merged**